model/riscv_insts_base.sail

```diff
@@ -321,28 +321,79 @@ function check_misaligned(vaddr : xlenbits, width : word_width) -> bool =
          WORD   => vaddr[0] == bitone | vaddr[1] == bitone,
          DOUBLE => vaddr[0] == bitone | vaddr[1] == bitone | vaddr[2] == bitone
        }
+function transform_VA (effective_address : xlenbits,PMLEN :int) -> xlenbits = {
+  assert(PMLEN >= 0, "PMLEN out of range");
+  assert(PMLEN <= 63, "PMLEN out of range");
+  let XLEN = sizeof(xlen);
+  let i = XLEN -(PMLEN +1);
+  let effective_bit = effective_address[i..i];
+  let NVBITS = replicate_bits(effective_bit, PMLEN);
+  let VBITS = effective_address[i..0];
+  NVBITS @ VBITS
+}
+
+function transform_PA (effective_address : xlenbits,PMLEN :int) -> xlenbits = {
+  assert(PMLEN >= 0, "PMLEN out of range");
+  assert(PMLEN <= 63, "PMLEN out of range");
+  let XLEN = sizeof(xlen);
+  let i = XLEN -(PMLEN + 1);
+  let NVBITS = replicate_bits(0b0, PMLEN);
+  let VBITS = effective_address[i..0];
+  NVBITS @ VBITS
+}
+
+function transform_effective_address(vaddr : xlenbits, pmm : bool) -> xlenbits = {
```

Comment:
For misaligned addresses, will this be applied to each byte of the address or just the initial address? (the spec mandates the former.)

Reply:
Currently, the  misaligned extension is not integrated into the Sail model. Therefore, I am applying Pointer Masking only to the initial address. However, once the integration is complete, we will need to apply Pointer Masking to each byte of the address.